### ISSUE TITLE
Build on Leonhard cluster

### ISSUE THREAD

**Issue by huanglangwen** [kosta777/parallel-genomeseq#22]
I have successfully built our program on the Leonhard cluster, here’s some note on how to successfully build it:
1. setup environment
```shell
module load gcc
module load openmpi
module load cmake
```
2. configure and make: DISABLE the `-flto` flag in CMakeLists.txt because gcc 8 can’t recognize it
```
mkdir build
cd build
cmake .. -DUSEMPI=ON -DUSEOMP=ON
make
```
@hanyao8 @spaceben @kosta777

**Comment by spaceben** [comment]
dont we have to use this: https://scicomp.ethz.ch/wiki/Using_the_batch_system

**Comment by huanglangwen** [comment]
Yes, but before we submit jobs to the batch system, we need to compile the program :)

**Comment by huanglangwen** [comment]
I found a good way to manage dependencies on clusters (including euler!): `spack`. With spack, it would be a piece of cake to handle our dependencies including C++17 compiler, openmpi, cmake and possibly HDF5 (with the chosen mpi). All we have to do is:
```bash
#module load python/3.6.0 #maybe have to update python
git clone https://github.com/spack/spack.git
cd spack/bin
./spack env activate myproject #bring installed packages into context, or manually load them using `./spack load <package>`
./spack install gcc@9.2.0
./spack install openmpi%gcc
./spack install hdf5%gcc
```

Spack will take a lot of time for compiling, but it should be all automatic.
